diff -burN binutils-2.25/config.sub binutils-2.25-ctr/config.sub
--- binutils-2.25/config.sub	2014-10-14 02:32:02.000000000 -0500
+++ binutils-2.25-ctr/config.sub	2015-05-14 12:18:03.000000000 -0500
@@ -594,6 +594,10 @@
 		basic_machine=crx-unknown
 		os=-elf
 		;;
+	ctr)
+		basic_machine=arm-none
+		os=-eabi
+		;;
 	da30 | da30-*)
 		basic_machine=m68k-da30
 		;;
diff -burN binutils-2.25/gold/binary.cc binutils-2.25-ctr/gold/binary.cc
--- binutils-2.25/gold/binary.cc	2014-10-14 02:32:04.000000000 -0500
+++ binutils-2.25-ctr/gold/binary.cc	2015-05-14 12:03:12.000000000 -0500
@@ -24,7 +24,11 @@
 
 #include <cerrno>
 #include <cstring>
+#ifdef __APPLE__
+#include <ctype.h>
+#else
 #include "safe-ctype.h"
+#endif
 
 #include "elfcpp.h"
 #include "stringpool.h"
@@ -153,7 +157,11 @@
   for (std::string::iterator p = mangled_name.begin();
        p != mangled_name.end();
        ++p)
+#ifdef __APPLE__
+    if (!isalnum(*p))
+#else
     if (!ISALNUM(*p))
+#endif
       *p = '_';
   mangled_name = "_binary_" + mangled_name;
   std::string start_symbol_name = mangled_name + "_start";
diff -burN binutils-2.25/gold/system.h binutils-2.25-ctr/gold/system.h
--- binutils-2.25/gold/system.h	2014-10-14 02:32:04.000000000 -0500
+++ binutils-2.25-ctr/gold/system.h	2015-05-14 12:04:56.000000000 -0500
@@ -110,6 +110,8 @@
 #define Unordered_map __gnu_cxx::hash_map
 #define Unordered_multimap __gnu_cxx::hash_multimap
 
+/* Clang libc++ support */
+#if !defined(_LIBCPP_VERSION)
 namespace __gnu_cxx
 {
 
@@ -130,6 +132,7 @@
 };
 
 }
+#endif /* !defined(_LIBCPP_VERSION) */
 
 #define reserve_unordered_map(map, n) ((map)->resize(n))
 
