diff -burN clang-3.6.0/CMakeLists.txt clang-3.6.0-ctr/CMakeLists.txt
--- clang-3.6.0/CMakeLists.txt	2015-01-05 08:06:14.000000000 -0600
+++ clang-3.6.0-ctr/CMakeLists.txt	2015-12-27 16:06:05.217982202 -0600
@@ -432,6 +432,10 @@
   message(FATAL_ERROR "Cannot disable static analyzer while enabling ARCMT")
 endif()
 
+option(CLANG_ENABLE_DIAGTOOL "Build diagtool." ON)
+option(CLANG_ENABLE_CLANGFORMAT "Build clang-format." ON)
+option(CLANG_ENABLE_CINDEXTEST "Build c-index-test." ON)
+
 if(CLANG_ENABLE_ARCMT)
   add_definitions(-DCLANG_ENABLE_ARCMT)
   add_definitions(-DCLANG_ENABLE_OBJC_REWRITER)
diff -burN clang-3.6.0/tools/CMakeLists.txt clang-3.6.0-ctr/tools/CMakeLists.txt
--- clang-3.6.0/tools/CMakeLists.txt	2014-07-16 11:48:33.000000000 -0500
+++ clang-3.6.0-ctr/tools/CMakeLists.txt	2015-12-27 16:06:32.546099110 -0600
@@ -1,9 +1,18 @@
-add_subdirectory(diagtool)
+if(CLANG_ENABLE_DIAGTOOL)
+  add_subdirectory(diagtool)
+endif()
+
 add_subdirectory(driver)
-add_subdirectory(clang-format)
-add_subdirectory(clang-format-vs)
 
-add_subdirectory(c-index-test)
+if(CLANG_ENABLE_CLANGFORMAT)
+  add_subdirectory(clang-format)
+  add_subdirectory(clang-format-vs)
+endif()
+
+if(CLANG_ENABLE_CINDEXTEST)
+  add_subdirectory(c-index-test)
+endif()
+
 add_subdirectory(libclang)
 
 if(CLANG_ENABLE_ARCMT)
