diff -burN llvm-3.6.0/cmake/config-ix.cmake llvm-3.6.0-ctr/cmake/config-ix.cmake
--- llvm-3.6.0/cmake/config-ix.cmake	2014-12-02 21:39:01.000000000 -0600
+++ llvm-3.6.0-ctr/cmake/config-ix.cmake	2015-12-27 15:23:15.770803157 -0600
@@ -499,9 +499,11 @@
 endif()
 
 set(LLVM_BINDINGS "")
-if(WIN32)
+
+if(LLVM_ENABLE_BINDINGS)
+  if(WIN32)
   message(STATUS "Go bindings disabled.")
-else()
+  else()
   find_program(GO_EXECUTABLE NAMES go DOC "go executable")
   if(GO_EXECUTABLE STREQUAL "GO_EXECUTABLE-NOTFOUND")
     message(STATUS "Go bindings disabled.")
@@ -515,13 +517,13 @@
       message(STATUS "Go bindings disabled, need at least Go 1.2.")
     endif()
   endif()
-endif()
+  endif()
 
-include(FindOCaml)
-include(AddOCaml)
-if(WIN32)
+  include(FindOCaml)
+  include(AddOCaml)
+  if(WIN32)
   message(STATUS "OCaml bindings disabled.")
-else()
+  else()
   find_package(OCaml)
   if( NOT OCAML_FOUND )
     message(STATUS "OCaml bindings disabled.")
@@ -539,6 +541,10 @@
       endif()
     endif()
   endif()
+  endif()
+else()
+  message(STATUS "Go bindings disabled.")
+  message(STATUS "OCaml bindings disabled.")
 endif()
 
 string(REPLACE " " ";" LLVM_BINDINGS_LIST "${LLVM_BINDINGS}")
diff -burN llvm-3.6.0/CMakeLists.txt llvm-3.6.0-ctr/CMakeLists.txt
--- llvm-3.6.0/CMakeLists.txt	2015-01-14 13:02:46.000000000 -0600
+++ llvm-3.6.0-ctr/CMakeLists.txt	2015-12-27 15:50:24.288257398 -0600
@@ -305,6 +305,8 @@
   "Build the LLVM tools. If OFF, just generate build targets." ON)
 
 option(LLVM_INCLUDE_UTILS "Generate build targets for the LLVM utils." ON)
+option(LLVM_BUILD_UTILS
+  "Build the LLVM utils. If OFF, just generate build targets." ON)
 
 option(LLVM_BUILD_RUNTIME
   "Build the LLVM runtime libraries." ON)
@@ -321,6 +323,8 @@
 option (LLVM_ENABLE_DOXYGEN "Use doxygen to generate llvm API documentation." OFF)
 option (LLVM_ENABLE_SPHINX "Use Sphinx to generate llvm documentation." OFF)
 
+option (LLVM_ENABLE_BINDINGS "Generate language bindings for Go, OCaml, etc." ON)
+
 option (LLVM_BUILD_EXTERNAL_COMPILER_RT
   "Build compiler-rt as an external project." OFF)
 
diff -burN llvm-3.6.0/tools/CMakeLists.txt llvm-3.6.0-ctr/tools/CMakeLists.txt
--- llvm-3.6.0/tools/CMakeLists.txt	2014-12-12 11:31:24.000000000 -0600
+++ llvm-3.6.0-ctr/tools/CMakeLists.txt	2015-12-27 15:41:20.233750881 -0600
@@ -16,7 +16,6 @@
 
 add_llvm_tool_subdirectory(opt)
 add_llvm_tool_subdirectory(llvm-as)
-add_llvm_tool_subdirectory(llvm-dis)
 add_llvm_tool_subdirectory(llvm-mc)
 
 add_llvm_tool_subdirectory(llc)
@@ -29,35 +28,9 @@
 add_llvm_tool_subdirectory(llvm-link)
 add_llvm_tool_subdirectory(lli)
 
-add_llvm_tool_subdirectory(llvm-extract)
-add_llvm_tool_subdirectory(llvm-diff)
-add_llvm_tool_subdirectory(macho-dump)
 add_llvm_tool_subdirectory(llvm-objdump)
 add_llvm_tool_subdirectory(llvm-readobj)
-add_llvm_tool_subdirectory(llvm-rtdyld)
 add_llvm_tool_subdirectory(llvm-dwarfdump)
-add_llvm_tool_subdirectory(dsymutil)
-add_llvm_tool_subdirectory(llvm-vtabledump)
-if( LLVM_USE_INTEL_JITEVENTS )
-  add_llvm_tool_subdirectory(llvm-jitlistener)
-else()
-  ignore_llvm_tool_subdirectory(llvm-jitlistener)
-endif( LLVM_USE_INTEL_JITEVENTS )
-
-add_llvm_tool_subdirectory(bugpoint)
-add_llvm_tool_subdirectory(bugpoint-passes)
-add_llvm_tool_subdirectory(llvm-bcanalyzer)
-add_llvm_tool_subdirectory(llvm-stress)
-add_llvm_tool_subdirectory(llvm-mcmarkup)
-
-add_llvm_tool_subdirectory(verify-uselistorder)
-
-add_llvm_tool_subdirectory(llvm-symbolizer)
-
-add_llvm_tool_subdirectory(llvm-c-test)
-
-add_llvm_tool_subdirectory(obj2yaml)
-add_llvm_tool_subdirectory(yaml2obj)
 
 add_llvm_tool_subdirectory(llvm-go)
 
@@ -71,6 +44,50 @@
 
 add_llvm_tool_subdirectory(gold)
 
+if(LLVM_BUILD_UTILS)
+  add_llvm_tool_subdirectory(llvm-dis)
+  add_llvm_tool_subdirectory(llvm-extract)
+  add_llvm_tool_subdirectory(llvm-diff)
+  add_llvm_tool_subdirectory(macho-dump)
+  add_llvm_tool_subdirectory(llvm-rtdyld)
+  add_llvm_tool_subdirectory(dsymutil)
+  add_llvm_tool_subdirectory(llvm-vtabledump)
+  if( LLVM_USE_INTEL_JITEVENTS )
+    add_llvm_tool_subdirectory(llvm-jitlistener)
+  else()
+    ignore_llvm_tool_subdirectory(llvm-jitlistener)
+  endif( LLVM_USE_INTEL_JITEVENTS )
+  add_llvm_tool_subdirectory(bugpoint)
+  add_llvm_tool_subdirectory(bugpoint-passes)
+  add_llvm_tool_subdirectory(llvm-bcanalyzer)
+  add_llvm_tool_subdirectory(llvm-stress)
+  add_llvm_tool_subdirectory(llvm-mcmarkup)
+  add_llvm_tool_subdirectory(verify-uselistorder)
+  add_llvm_tool_subdirectory(llvm-symbolizer)
+  add_llvm_tool_subdirectory(llvm-c-test)
+  add_llvm_tool_subdirectory(obj2yaml)
+  add_llvm_tool_subdirectory(yaml2obj)
+else()
+  ignore_llvm_tool_subdirectory(llvm-dis)
+  ignore_llvm_tool_subdirectory(llvm-extract)
+  ignore_llvm_tool_subdirectory(llvm-diff)
+  ignore_llvm_tool_subdirectory(macho-dump)
+  ignore_llvm_tool_subdirectory(llvm-rtdyld)
+  ignore_llvm_tool_subdirectory(dsymutil)
+  ignore_llvm_tool_subdirectory(llvm-vtabledump)
+  ignore_llvm_tool_subdirectory(llvm-jitlistener)
+  ignore_llvm_tool_subdirectory(bugpoint)
+  ignore_llvm_tool_subdirectory(bugpoint-passes)
+  ignore_llvm_tool_subdirectory(llvm-bcanalyzer)
+  ignore_llvm_tool_subdirectory(llvm-stress)
+  ignore_llvm_tool_subdirectory(llvm-mcmarkup)
+  ignore_llvm_tool_subdirectory(verify-uselistorder)
+  ignore_llvm_tool_subdirectory(llvm-symbolizer)
+  ignore_llvm_tool_subdirectory(llvm-c-test)
+  ignore_llvm_tool_subdirectory(obj2yaml)
+  ignore_llvm_tool_subdirectory(yaml2obj)
+endif()
+
 add_llvm_external_project(clang)
 add_llvm_external_project(llgo)
 
